hummingbird/views.py

```diff
@@ -0,0 +1,195 @@
+		# Populate forms
+		usersong_form = UserSongForm()
+		device_form = UserDeviceForm()
+		context_dict['usersong_form'] = usersong_form
+		context_dict['device_form'] = device_form
+	except UserProfile.DoesNotExist:
+		pass
+	# If we updated one of the fields, pass that to the template so it can tell the user that update was successful.
+	context_dict['added_device'] = added_device
+	context_dict['added_song'] = added_song
+	return render(request,'profile.html',context_dict)
+
+
+## Checks if user exists. If so, return user name. Otherwise, return "0". 
+def get_user_from_device(request):
+	if request.method == 'GET':
+		try:
+			device = request.GET['mac_id']
+			userdevice = UserDevice.objects.get(mac_id=device.lower())
+			user = userdevice.user_profile.id
+			return HttpResponse(user, content_type='text/plain')
+		except UserDevice.DoesNotExist:
+			return HttpResponse("0", content_type='text/plain')
+	pass
+
+
+## Returns information necessary to build the User object if mac_id is recognized. If not, return "0".
+## If there is a user, a string representation of a dictioanry is returned.
+def build_user_from_device(request):
+	if request.method == 'GET':
+		try:
+			device=urllib.unquote(request.GET['mac_id']).decode('utf8')
+			userdevice = UserDevice.objects.get(mac_id=device.lower()) 
+			user_dict={}
+			user_dict['name'] = userdevice.user_profile.name
+			# If user exists, but there is no song currently associated with them, we don't have anything to play,
+			# so we return "0"
+			if userdevice.user_profile.song.name=='':
+				return HttpResponse("0", content_type='text/plain') 
+			else:
+				user_dict['song'] = userdevice.user_profile.song.name
+			user_dict['length'] = userdevice.user_profile.length
+			# Handle first-seen case. This will likely happen right after the device is added.
+			# So, to prevent immediate playback after adding a user and device, if this is the 
+			# first time a user is seen, just set the "last_played" field to now, and they will play tomorrow.
+			if userdevice.user_profile.last_played is not None:
+				user_dict['last_played'] = userdevice.user_profile.last_played.strftime('%Y-%m-%d %H:%M:%S')
+			else: user_dict['last_played'] = datetime.datetime(1991,1,1).strftime('%Y-%m-%d %H:%M:%S')
+			return HttpResponse(str(user_dict), content_type='text/plain')
+		except UserDevice.DoesNotExist:
+			return HttpResponse("0", content_type='text/plain')
+	pass
+
+
+## Once a user's song is played, Hummingbird uses this to update their "last_played" field to the current datetime.
+def update_last_played(request):
+	if request.method == 'GET':
+		device=urllib.unquote(request.GET['mac_id']).decode('utf8')
+		userdevice = UserDevice.objects.get(mac_id=device)
+		userprofile = userdevice.user_profile
+		userprofile.last_played = datetime.datetime.now()
+		userprofile.save()
+		return HttpResponse("Updated User", content_type='text/plain')
+	else:
+		return HttpResponse("0", content_type='text/plain')
```

Comment:
Yep, that's fine. For reference, here's the Trello card: https://trello.com/c/AIb94IKx